From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Herman van Hazendonk <github.com@herrie.org>
Date: Thu, 29 Oct 2026 16:01:33 +0100
Subject: [PATCH] PmsLogging: pass the formatted message as an argument, not as
 the format

PmsDebugLog(), PmsNoticeLog(), PmsInfoLog(), PmsWarningLog() and
PmsErrorLog() format the caller's message into a buffer with vsnprintf()
and then hand that buffer to Logger::Debug() and friends, which are
printf-style and take it as the format string. Any '%' that reaches the
message - from a client name, a payload or an error string - is then
interpreted a second time and reads arguments that were never passed.

Log the buffer with "%s". With this, the header builds with
-Werror=format-security, which oe-core turns on by default.

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
Upstream-Status: Pending
---
 include/public/pmscore/PmsLogging.h | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/include/public/pmscore/PmsLogging.h b/include/public/pmscore/PmsLogging.h
index d5a42f5..8ed3371 100644
--- a/include/public/pmscore/PmsLogging.h
+++ b/include/public/pmscore/PmsLogging.h
@@ -60,7 +60,7 @@ public:
             int len = snprintf(buffer, sizeof(buffer), "%s::%s ", PmsClass, PmsFunc);
             len = vsnprintf(&buffer[len], sizeof(buffer) - len, format, args);
 
-            pLogger->Debug(buffer);
+            pLogger->Debug("%s", buffer);
 
             va_end (args);
         }
@@ -76,7 +76,7 @@ public:
             int len = snprintf(buffer, sizeof(buffer), "%s::%s ", PmsClass, PmsFunc);
             len = vsnprintf(&buffer[len], sizeof(buffer) - len, format, args);
 
-            pLogger->Notice(buffer);
+            pLogger->Notice("%s", buffer);
 
             va_end (args);
         }
@@ -92,7 +92,7 @@ public:
             int len = snprintf(buffer, sizeof(buffer), "%s::%s ", PmsClass, PmsFunc);
             len = vsnprintf(&buffer[len], sizeof(buffer)-len, format, ar);
 
-            pLogger->Info(buffer);
+            pLogger->Info("%s", buffer);
 
             va_end (ar);
         }
@@ -108,7 +108,7 @@ public:
             int len = snprintf(buffer, sizeof(buffer), "%s::%s ", PmsClass, PmsFunc);
             len = vsnprintf(&buffer[len], sizeof(buffer)-len, format, ar);
 
-            pLogger->Warning(buffer);
+            pLogger->Warning("%s", buffer);
 
             va_end (ar);
         }
@@ -125,7 +125,7 @@ public:
             int len = snprintf(buffer, sizeof(buffer), "%s::%s Error=%s(%d) ", PmsClass, PmsFunc, errString, err);
             len = vsnprintf(&buffer[len], sizeof(buffer)-len, format, ar);
 
-            pLogger->Error(buffer);
+            pLogger->Error("%s", buffer);
 
             va_end (ar);
         }
